### ISSUE TITLE
Push fails with "Cannot read property 'success' of undefined" error when Apex file names do not match

### ISSUE THREAD

**Issue by pozil** [forcedotcom/cli#674]
### Summary

When Apex class `.cls` and `.cls-meta.xml` file names do not match, CLI push will fail with the following error:
```
ERROR running force:source:push:  Cannot read property 'success' of undefined
```

### Steps To Reproduce:

```
sfdx force:org:create -s -a demo -f config/project-scratch-def.json
cd demo
sfdx force:apex:class:create -n Demo -d force-app/main/default/classes
mv force-app/main/default/classes/Hello.cls force-app/main/default/classes/HelloFAIL.cls
sfdx force:source:push
```

### Expected result

We should get a clear error message telling us that the Apex file names do not match.

### Actual result

`sfdx force:source:push` command fails with:
```
ERROR running force:source:push:  Cannot read property 'success' of undefined
```

### Additional information

I could reproduce the issue on the release and prerelease of the CLI.

**Comment by kvor** [comment]
Same thing happens when `.cls-meta.xml` is missing

**Comment by shetzel** [comment]
I have a fix for this in review and assuming everything checks out it will be included in the release this Saturday.  Thanks @pozil

**Comment by DataBowman** [comment]
I've had this issue since Oct 13th 

"Cannot read property 'success' of undefined"
<img width="455" alt="Screenshot 2020-10-15 at 09 45 12" src="https://user-images.githubusercontent.com/17692441/96431327-4068b700-11fb-11eb-94a6-ff42f9e6ff91.png">


<img width="587" alt="Screenshot 2020-10-13 at 20 51 23" src="https://user-images.githubusercontent.com/17692441/96431350-49f21f00-11fb-11eb-8b4a-937bd95b93ff.png">

ERROR running force:source:deploy:  Cannot read property 'success' of undefined

**Comment by shetzel** [comment]
This is now fixed with CLI version 7.76.1 and salesforcedx plugin version 50.1.1
